File: backend/app/enums.py
"""
Перечисления (enums) для всего приложения.
"""
from enum import Enum


class StudentStatus(str, Enum):
    """Статус студента"""
    STUDY = "study"
    EXPELLED = "expelled"
    ACADEMIC_LEAVE = "academic_leave"


class OrderType(str, Enum):
    """Тип приказа"""
    ENROLLMENT = "enrollment"
    EXPULSION = "expulsion"
    NEXT_COURSE = "next_course"
    ACADEMIC_LEAVE = "academic_leave"
    TRANSFER_DIRECTION = "transfer_direction"


class ExpulsionReason(str, Enum):
    """Причина отчисления"""
    DEBTS = "debts"
    HEALTH = "health"
    PERSONAL_REASON = "personal_reason"
    OWN = "own"


class AcademicLeaveReason(str, Enum):
    """Причина академического отпуска"""
    HEALTH = "health"
    PERSONAL_FAMILY = "personal/family"


class ControlType(str, Enum):
    """Тип контроля дисциплины"""
    TEST = "test"
    EXAM = "exam"
    COURSE_WORK = "course work"


class EducationForm(str, Enum):
    """Форма обучения"""
    FULL_TIME = "full-time"
    PART_TIME = "part-time"
    EXTRAMURAL = "extramural"

File: frontend/UniversityContingent/Models/Enums.cs
namespace UniversityContingent.Models
{
    /// <summary>
    /// Статус студента
    /// </summary>
    public enum StudentStatus
    {
        study,              // Обучается
        expelled,           // Отчислен
        academic_leave      // Академический отпуск
    }

    /// <summary>
    /// Тип приказа
    /// </summary>
    public enum OrderType
    {
        enrollment,         // Зачисление
        expulsion,          // Отчисление
        next_course,        // Перевод на следующий курс
        academic_leave,     // Предоставление академического отпуска
        transfer_direction  // Перевод на другое направление
    }

    /// <summary>
    /// Причина отчисления
    /// </summary>
    public enum ExpulsionReason
    {
        debts,          // Академическая задолженность
        health,         // По состоянию здоровья
        personal_reason,// Личные обстоятельства
        own             // По собственному желанию
    }

    /// <summary>
    /// Причина академического отпуска
    /// </summary>
    public enum AcademicLeaveReason
    {
        health,         // По состоянию здоровья
        personal_family // По личным/семейным обстоятельствам
    }

    /// <summary>
    /// Тип контроля
    /// </summary>
    public enum ControlType
    {
        test,           // Зачёт
        exam,           // Экзамен
        course_work     // Курсовая работа
    }

    /// <summary>
    /// Форма обучения
    /// </summary>
    public enum EducationForm
    {
        full_time,      // Очная
        part_time,      // Заочная
        extramural      // Очно-заочная
    }
}
